### PR TITLE
Remove FullInvocationId as dependency to send invocation responses

### PR DESCRIPTION
**Pull request by slinkydeveloper**
See `create_response_message` signature. This removes the need to store additional data when implementing #1324

### CODE REVIEW

File: cli/src/commands/invocations/describe.rs [hunk 1]
```diff
@@ -100,14 +100,7 @@ async fn describe(env: &CliEnv, opts: &Describe) -> Result<()> {
...
     if let Some(parent) = &inv.invoked_by_id {
-        c_println!(
-            "{} {}",
-            inv.invoked_by_component
-                .as_ref()
-                .map(|x| style(x.to_owned()).italic().blue())
-                .unwrap_or_else(|| style("<UNKNOWN>".to_owned()).red()),
-            style(parent).italic(),
-        );
+        c_println!("{}", style(parent).italic());
```
Comment (slinkydeveloper) [inline cli/src/commands/invocations/describe.rs:103]: If we really need this info back in the CLI, we could reintroduce it later with a join in the query.

File: crates/storage-proto/src/lib.rs [hunk 2]
```diff
@@ -441,7 +448,7 @@ pub mod storage {
                     let source = match value {
                         restate_types::invocation::Source::Ingress => source::Source::Ingress(()),
                         restate_types::invocation::Source::Service(fid) => {
```
Comment (tillrohrmann) [inline crates/storage-proto/src/lib.rs:450]: This is no longer the `fid`.

File: crates/types/src/invocation.rs [hunk 3]
```diff
@@ -198,8 +198,7 @@ pub enum ServiceInvocationResponseSink {
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum Source {
     Ingress,
-    // TODO This can be changed to simply InvocationId
-    Service(FullInvocationId),
+    Service(InvocationId),
```
Comment (tillrohrmann) [inline crates/types/src/invocation.rs:201]: Can we also capture the service name? Then we wouldn't have to do the changes on the CLI side. Knowing not only the invocation id but also the service name can be useful information for users.

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @slinkydeveloper. I think it makes sense to remove the fid from the invocation response. One thing I am wondering is whether `Source::Service` could still carry the service name. Would the service name be available at the creation sites of `Source::Service` in the `fid-less` world?

File: crates/types/src/invocation.rs (hunk 3 above)
Comment (slinkydeveloper) [reply crates/types/src/invocation.rs:201]: > Would the service name be available at the creation sites of Source::Service in the fid-less world?

No. In my PR about idempotency, i would need to store additional data to reconstruct that, and I don't want to do it, as this diagnostic info can be retrieved anyway differently.

Comment (slinkydeveloper) [reply crates/types/src/invocation.rs:201]: Essentially here https://github.com/restatedev/restate/pull/1326/files#diff-3c9cbf4d617758ec9be8fb418fa90278ac98881145eea5a2bced65a2e417978bR75

I would need to store the additional metadata of the source only for this purpose.

File: cli/src/commands/invocations/describe.rs (hunk 1 above)
Comment (AhmedSoliman) [reply cli/src/commands/invocations/describe.rs:103]: It was really useful, if anything, we want to improve observability of invocation trees.

File: crates/types/src/invocation.rs (hunk 3 above)
Comment (slinkydeveloper) [reply crates/types/src/invocation.rs:201]: If we really feel this is a problem, we could re-introduce this later once we sort out all the details of #1329. I also think this is a relative problem of the CLI: in a webui, probably every `invocation_id` will show up as a link to a page showing the invocation status, so the navigation will be much easier in that sense that in CLI.

Comment (slinkydeveloper) [reply crates/types/src/invocation.rs:201]: One second i might have overlooked something, i might have a way to preserve this fid.

Comment (tillrohrmann) [reply crates/types/src/invocation.rs:201]: > i might have a way to preserve this fid.

That be great :-)

File: crates/worker/src/partition/types.rs [hunk 4]
```diff
@@ -90,7 +89,7 @@ pub fn create_response_message(
                     caller_context,
                 }
                 .encode_to_vec(),
-                Source::Service(callee.clone()),
+                Source::Internal,
```
Comment (slinkydeveloper) [inline crates/worker/src/partition/types.rs:92]: By taking this shortcut, we can preserve the `Source::Service`. This only affects internal callbacks (which we'll stop using anyway now that i'm reimplementing idempotency)

**Pull request merged**